### PR TITLE
Add ability to read auth from file

### PR DESCRIPTION
**Pull request by proudmuslim-dev**
Usage: 

```
$ miniserve --auth-file auth.txt
```

Example auth.txt content

```
joe:123
bob:sha256:a665a45920422f9d417e4867efdc4fb8a04a1f3fff1fa07e998e86f7f7a27ae3
bill:
```

Related issue: #980

### CODE REVIEW

File: src/args.rs [hunk 1]
```diff
@@ -72,6 +72,10 @@ pub struct CliArgs {
     )]
     pub auth: Vec<auth::RequiredAuth>,
 
+    /// Read authentication values from a file.
```
Comment (svenstaro) [inline src/args.rs:75]:
```suggestion
    /// Read authentication values from a file
```

Comment (svenstaro) [reply src/args.rs:75]: Also please describe the format of the file.

**Review by svenstaro — CHANGES_REQUESTED** [review]
Very cool, I think this would be a good addition. Please fix my comments and then make sure to also update the README in a separate commit.

**Comment by svenstaro** [comment]
Are you still working on this?

**Comment by proudmuslim-dev** [comment]
I would have squashed some of those commits, but I don't currently have my SSH or my GPG keys set up so I'm limited to the web interface. Sorry about that :sweat_smile:

**Review by svenstaro — CHANGES_REQUESTED** [review]
Sorry I didn't say this earlier but I'd like to see a test for this. It's a security feature so it should definitely be tested.

**Comment by proudmuslim-dev** [comment]
Alright, just give me some time

**Comment by svenstaro** [comment]
Thanks, great work! I'll merge this and clean it up a little but I don't want to drag merging this out further.

**Comment by proudmuslim-dev** [comment]
Anytime! I realize now that the tests could've been done better, apologies

**Comment by svenstaro** [comment]
If you'd like to clean it up in some fashion, I'll surely accept a follow-up!

**Pull request merged**